File: 908-MiddleOfTheLinkedList/908-MiddleOfTheLinkedList.py
# Last updated: 10/8/2025, 9:49:42 PM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        middle=head
        end=head
        while end and end.next:
            middle=middle.next
            end=end.next.next
        return middle

        

File: 908-MiddleOfTheLinkedList/908-MiddleOfTheLinkedList.java
// Last updated: 10/8/2025, 9:49:44 PM
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
      ListNode slow=head;
        ListNode fast=head;
        //[1,2,3,4,5]
        while(fast!=null && fast.next!=null){
            slow=slow.next;
            fast=fast.next.next;
        }
        return slow;  
    }
}